quipucords/tests/integration/test_network_scan.py

```diff
@@ -225,6 +228,48 @@ def fingerprint_fact_map():
     }
 
 
+@pytest.fixture
+def unexpected_fingerprints():
+    """Set of facts that wouldn't appear on network scans."""
+    return {
+        # sattelite / vcenter exclusive
+        "virtual_host_name",
+        "virtual_host_uuid",
+        # vcenter exclusive
+        "vm_cluster",
+        "vm_datacenter",
+        "vm_dns_name",
+        "vm_host_core_count",
+        "vm_host_socket_count",
+        "vm_state",
+        "vm_uuid",
+    }
+
+
+def test_sanity_check_raw_fact_matches(
+    expected_network_scan_facts, fingerprint_fact_map, unexpected_fingerprints
+):
+    """Ensure raw facts mapped to fingerprint facts match known facts."""
+    raw_facts_for_fingerprints = reduce(
```

Comment:
I stumbled upon this [article](https://www.artima.com/weblogs/viewpost.jsp?thread=98196) written by Guido saying that instead of using reduce(), the more "pythonic" way is to implement an accumulating function the old fashioned way. 
I honestly think reduce() is easier to read and "cleaner", but thought about sharing this with you.

Reply:
Oh, I remember this article. There's a reason for "hiding" reduce under functools (back in python 2 era reduce didn't need import).

I can rewrite this w/o reduce and we check which version has better readability